torch/functional.py

```diff
@@ -361,7 +370,16 @@ def parse_subscript(n: int) -> str:
         # in the original implementation this line is omitted
         return einsum(equation, *_operands)
 
-    return _VF.einsum(equation, operands)  # type: ignore[attr-defined]
+    if len(operands) <= 2:
+        # the path for contracting 0 or 1 time(s) is already optimized
+        return _VF.einsum(equation, operands)  # type: ignore[attr-defined]
+
+    path = None
+    if _opt_einsum is not None:
+        tupled_path = _opt_einsum.contract_path(equation, *operands)[0]
```

Comment:
Do we want to consider caching this result based on the equation and operands sizes?

Reply:
oooh yes, will explore this some more

Reply:
I played with caching the path + it sadly doesn't seem like a good idea given our benchmarks + the fact that people don't usually call einsum with the same equation + tensor order. If we want to truly take advantage of caching, we should explore common patterns + go more deeply than just caching for size, which should would not be in scope of this PR.

Reply:
FYI the benchmarks above are benchmarked with code in https://github.com/pytorch/pytorch/pull/84890/commits/1d1fa9c15065e23dee23900127eb5fcc4e471b9b, except i actually called the memoized `_contract_path` (the commit does not)

Reply:
(newly https://github.com/pytorch/pytorch/pull/84890/commits/be236c7ea889bee9f727262ebc0464132c5cc51d)